atlasdb-config/src/test/java/com/palantir/atlasdb/SinglePingableLeaderTest.java

```diff
@@ -0,0 +1,143 @@
+    private static final MappingBuilder PING_V2_MAPPING = WireMock.get(WireMock.urlEqualTo(PING_V2));
+    private static final MappingBuilder UUID_MAPPING = WireMock.get(WireMock.urlEqualTo(UUID_PATH));
+
+    private static final SslConfiguration SSL_CONFIGURATION
+            = SslConfiguration.of(Paths.get("var/security/trustStore.jks"));
+
+    private int availablePort;
+    private String serverUri;
+
+    @Rule
+    public WireMockRule availableServer = new WireMockRule(WireMockConfiguration.wireMockConfig().dynamicPort());
+
+    @Before
+    public void setup() {
+        availableServer.stubFor(UUID_MAPPING.willReturn(aResponse().withStatus(200).withBody(
+                ("\"" + REMOTE_UUID.toString() + "\"").getBytes())));
+
+        availableServer.stubFor(PING_MAPPING.willReturn(WireMock.aResponse().withStatus(204)));
+
+        availablePort = availableServer.port();
+        serverUri = String.format("http://%s:%s", WireMockConfiguration.DEFAULT_BIND_ADDRESS, availablePort);
+    }
+
+    @Test
+    public void doesNotFallbackOnPingWhenPingV2Responds() {
+        availableServer.stubFor(PING_V2_MAPPING.willReturn(WireMock.aResponse().withStatus(200)
+                .withBody("{\"isLeader\":false}")));
+        pingerWithVersion(OrderableSlsVersion.valueOf("1.1.1")).pingLeaderWithUuid(REMOTE_UUID);
+        List<LoggedRequest> requests = WireMock.findAll(WireMock.getRequestedFor(WireMock.urlMatching(PING_V2)));
+        assertThat(requests.size()).isEqualTo(1);
+        requests = WireMock.findAll(WireMock.getRequestedFor(WireMock.urlMatching(PING)));
+        assertThat(requests.size()).isEqualTo(0);
+    }
+
+    @Test
+    public void fallbackOnPingWhenPingV2DoesNotExist() {
+        availableServer.stubFor(PING_V2_MAPPING.willReturn(WireMock.aResponse().withStatus(404)));
+        verifyPingRequests(5, 5);
+    }
+
+    @Test
+    public void fallbackOnPingWhenPingV2Throws() {
+        availableServer.stubFor(PING_V2_MAPPING.willReturn(WireMock.aResponse().withStatus(500)));
+        verifyPingRequests(5, 5);
+    }
+
+    @Test
+    public void samplesRequestsToFailingPing2() {
+        availableServer.stubFor(PING_V2_MAPPING.willReturn(WireMock.aResponse().withStatus(500)));
+        verifyPingRequests(100, 100);
+    }
+
+    public void verifyPingRequests(int totalNumOfPings, int verifyLegacyPingRequests) {
+        LeaderPinger pinger = pingerWithVersion(OrderableSlsVersion.valueOf("1.1.1"));
+        IntStream.range(0, totalNumOfPings).forEach(idx -> pinger.pingLeaderWithUuid(REMOTE_UUID));
+
+        List<LoggedRequest> requests = WireMock.findAll(WireMock.getRequestedFor(WireMock.urlMatching(PING_V2)));
+        assertThat(requests.size()).isLessThan(totalNumOfPings);
+        requests = WireMock.findAll(WireMock.getRequestedFor(WireMock.urlMatching(PING)));
+        assertThat(requests.size()).isEqualTo(verifyLegacyPingRequests);
```

Comment:
I would suggest having this be two methods: one for `assertLegacyPingRequestsMade(int)` and one for `assertPingV2RequestsMadeAreAtMost(int)`. The 2 numbers is a bit messy to read.